### PR TITLE
repo-updater: Syncing improvements for OTHER external services

### PR DESCRIPTION
**Pull request by tsenart**
This commit integrates the OTHER external service kind with the new
Syncer. This was required to do in order to make the scheduler handle
renames because the `OtherExternalServicesSyncer` emitted
`protocol.RepoInfo` which doesn't have an ID.

Populating it's ID would require changes that we wouldn't be using in
the long run, so I gave it a shot to integrate this with the new Syncer.

Notably, the syncer won't handle renames of repos of this
kind since we don't have a code host API as a source of truth with a stable identifier.

Part of #2025

### CODE REVIEW

**Comment by tsenart** [comment]
@keegancsmith: You had mentioned some concerns in the past about this external service kind together with the new syncer. Do you remember what those concerns were?

File: cmd/repo-updater/repos/sources.go [hunk 1]
```diff
@@ -324,3 +328,97 @@ func bitbucketserverRepoToRepo(
...
+type OtherSource struct {
+	svc  *ExternalService
+	conn *schema.OtherExternalServiceConnection
+}
+
+// NewOtherSource returns a new OtherSource from the given external service.
+func NewOtherSource(svc *ExternalService) (*OtherSource, error) {
+	var c schema.OtherExternalServiceConnection
+	if err := jsonc.Unmarshal(svc.Config, &c); err != nil {
+		return nil, fmt.Errorf("external service id=%d config error: %s", svc.ID, err)
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/sources.go]:
```suggestion
		return nil, errors.Wrapf(err, "external service id=%d config error", svc.ID)
```

File: cmd/repo-updater/repos/sources.go [hunk 2]
```diff
@@ -324,3 +328,97 @@ func bitbucketserverRepoToRepo(
...
+			return nil, err
+		}
+		parseRepo = baseURL.Parse
+	}
+
+	cloneURLs := make([]*url.URL, 0, len(s.conn.Repos))
+	for _, repo := range s.conn.Repos {
+		cloneURL, err := parseRepo(repo)
+		if err != nil {
+			log15.Error("skipping invalid repo clone URL", "repo", repo, "url", s.conn.Url, "error", err)
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/sources.go]: should we maybe hard fail this so the user gets feedback in the external service editor?

Comment (tsenart) [reply cmd/repo-updater/repos/sources.go]: Yep. This now makes sense to do.

**Pull request merged**